fvm/evm/offchain/utils/verify.go

```diff
@@ -0,0 +1,293 @@
+package utils
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/onflow/cadence"
+	"github.com/onflow/cadence/encoding/ccf"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/fvm/environment"
+	"github.com/onflow/flow-go/fvm/evm"
+	"github.com/onflow/flow-go/fvm/evm/events"
+	"github.com/onflow/flow-go/ledger"
+	"github.com/onflow/flow-go/ledger/common/convert"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/executiondatasync/execution_data"
+	"github.com/onflow/flow-go/storage"
+)
+
+// EVM Root Height is the first block that has EVM Block Event where the EVM block height is 1
+func IsEVMRootHeight(chainID flow.ChainID, flowHeight uint64) bool {
+	if chainID == flow.Testnet {
+		return flowHeight == 211176670
+	} else if chainID == flow.Mainnet {
+		return flowHeight == 85981135
```

Comment:
I believe that the proper Flow Height should be `85981134`. This is where the `EVM` contract was first deployed on mainnet, see https://www.flowscan.io/contract/A.e467b9dd11fa00df.EVM/deployments?deployment=85981134

Reply:
This height is used to skip the root height where there is no execution data to read EVM events from. 

Technically, you are right, but I can't use `85981134` with mainnet25's data, because `85981134` is a block from mainnet24. And this doesn't matter much as both `85981134` and `85981135` have no EVM events. 

It only matters if we are verifying the state change at register level, which is not the case at the moment.